File: InferenceEngine/probrecord.cpp
#include "StdAfx.h"
#include "probrecord.h"

using namespace std;
/*
 * Author: Sari Haj Hussein
 */
/** public methods **/
void probrecord::info()
{
	cout << "General Prob Record Information" << endl;
	cout << "-------------------------------" << endl;
	cout << "License Plate: " << this->licensePlate << endl;
	cout << "Prob Location: " << this->probLocation << endl;
	cout << "Start Time: " << this->sTime << endl;
	cout << "End Time: " << this->eTime << endl;
	cout << "Start Time Unix: " << this->sTimeUnix << endl;
	cout << "End Time Unix: " << this->eTimeUnix << endl;
}

string probrecord::getLicensePlate()
{
	return this->licensePlate;
}

string probrecord::getProbLocation()
{
	return this->probLocation;
}

string probrecord::getsTime()
{
	return this->sTime;
}

string probrecord::geteTime()
{
	return this->eTime;
}

__int64 probrecord::getsTimeUnix()
{
	return this->sTimeUnix;
}

__int64 probrecord::geteTimeUnix()
{
	return this->eTimeUnix;
}

/** constructor **/
probrecord::probrecord(string licensePlate, string probLocation, string sTime, string eTime, __int64 sTimeUnix, __int64 eTimeUnix)
{
	this->licensePlate = licensePlate;
	this->probLocation = probLocation;
	this->sTime = sTime;
	this->eTime = eTime;
	this->sTimeUnix = sTimeUnix;
	this->eTimeUnix = eTimeUnix;
}

/** destructor **/
probrecord::~probrecord()
{
}

File: src/info/sarihh/unimodeling/utility/ProbRecord.java
package info.sarihh.unimodeling.utility;

import java.sql.Timestamp;
import java.text.SimpleDateFormat;
import java.util.Locale;

/*
 * Author: Sari Haj Hussein
 */
public class ProbRecord {

    public ProbRecord(String licensePlate, String probLoc, Timestamp sTime, Timestamp eTime, long sTimeUnix, long eTimeUnix) {
        this.licensePlate = licensePlate;
        this.probLoc = probLoc;
        this.sTime = sTime;
        this.eTime = eTime;
        this.sTimeUnix = sTimeUnix;
        this.eTimeUnix = eTimeUnix;
    }

    public String getLicensePlate() {
        return licensePlate;
    }

    public String getProbLoc() {
        return probLoc;
    }

    public Timestamp getsTime() {
        return sTime;
    }

    public Timestamp geteTime() {
        return eTime;
    }

    public long getsTimeUnix() {
        return sTimeUnix;
    }

    public long geteTimeUnix() {
        return eTimeUnix;
    }

    public String getsTimeString() {
        return dateFormat.format(sTime).replace(" ", "!");
    }

    public String geteTimeString() {
        return dateFormat.format(eTime).replace(" ", "!");
    }
    String licensePlate, probLoc;
    Timestamp sTime, eTime;
    long sTimeUnix, eTimeUnix;
    private static SimpleDateFormat dateFormat =
            new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSSSSSSS", Locale.ENGLISH);
}